src/nested_pandas/nestedframe/core.py

```diff
@@ -518,18 +519,37 @@ def query(self, expr: str, *, inplace: bool = False, **kwargs) -> NestedFrame |
         # to the nest and repack.  Otherwise, apply it to this instance as usual,
         # since it operated on the base attributes.
         if isinstance(result, _SeriesFromNest):
+            if not is_bool_dtype(result.dtype):
+                raise ValueError("Query condition must evaluate to a boolean Series")
```

Comment:
Adding a simple test to trigger this has yielded some odd and unexpected errors:

```python
base.query("a + 2 / b")
```
does not yield this `ValueError`, but a `KeyError` complaining that some array (it shows the values) does not exist in `[index]`.  Still looking into this, but it appears some index plumbing is amiss.

Reply:
We don’t want to support this index functionality, at least not right now. Thank you for investigating to provide a better error message!

Reply:
Verified just now that this same problem afflicts vanilla Pandas DataFrame objects, so it's not related to our changes and is probably out of scope here.  It does make it harder to trigger the new `ValueError`; maybe that's what we should scale back.